src/ucs/async/async.c

```diff
@@ -414,7 +414,7 @@ ucs_async_alloc_handler(int min_id, int max_id, ucs_async_mode_t mode,
     ucs_free(handler);
 err_dec_num_handlers:
     if (async != NULL) {
-        ucs_atomic_add32(&async->num_handlers, -1);
+        ucs_atomic_add32(&async->num_handlers, (uint32_t)-1);
```

Comment:
maybe add helper functions ucs_atomic_addi32 and ucs_atomic_faddi32 to take int32_t ?